src/core_plugins/kibana/public/context/app.js

```diff
@@ -42,10 +42,13 @@ module.directive('contextApp', function ContextApp() {
   };
 });
 
-function ContextAppController($scope, config, Private) {
+function ContextAppController($scope, config, Private, timefilter) {
   const queryParameterActions = Private(QueryParameterActionsProvider);
   const queryActions = Private(QueryActionsProvider);
 
+  // this is apparently the "canonical" way to disable the time picker
+  timefilter.enabled = false;
```

Comment:
huh... that concerning. As you say, this seems to be the way to do it at the moment though.